portfolio/src/main/webapp/assets/js/script.js

```diff
@@ -99,4 +100,72 @@ function toggleImageGallery() {
     toggleButton.innerHTML = "view image gallery"
 
   }
+}
+
+function getComments() {
+    fetch("/data").then(response => response.json()).then((comments) => {
+        console.log(comments);
+        const commentsContainer = document.getElementById('comments-container');
+        comments.map((comment) => commentsContainer.appendChild(createCommentSection(comment)));
+
+  });
+}
+
+/** Creates some HTML that displays the comments 
+    A comment in HTML will look like below:
+    <div id="comment-wrapper">
+        <div id="comment-header">
+            <p id="username-display">daliscious123</p>
+            <p id="timestamp-display">Wed 10/06/2020 00:11:54 BNT</p>
+        </div>
+        <p id="comments-display">
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+        eiusmod tempor incididunt ut labore et dolore magna aliqua.
+        Dapibus ultrices in iaculis nunc sed augue lacus. Quam nulla
+        porttitor massa id neque aliquam. Ultrices mi tempus imperdiet
+        nulla malesuada. Eros in cursus turpis massa tincidunt dui ut
+        ornare lectus.
+        </p>
+        <hr class="solid" />
+    </div>
+*/
+function createCommentSection(commentObj) {
+
+  // comment wrapper div
+  const commentWrapper = document.createElement("div");
+  commentWrapper.id = "comment-wrapper";
+  // username and timestamp wrapper div
+  const headerWrapper = document.createElement("div");
```

Comment:
Instead of creating each of the child nodes manually, you can just set:

`commentWrapper.innerHTML = '<div id="comment-header">
            <p id="username-display"></p>
            <p id="timestamp-display"></p>
        </div>
        <p id="comments-display"></p>
        <hr class="solid" />'`

Then you can use `document.getElementById("username-display").innerHTML = comment.username`, and similar for the other elements.

Reply:
thanks! this is cool

Reply:
hey Cristen! The above didnt work because I think it isn't loaded in the document yet so i did this instead:

  commentWrapper.innerHTML =
    `<div id="comment-header"> <p id="username-display">${comment.username}</p> <p id="timestamp-display">${comment.timestamp}</p> </div> <p id="comments-display">${comment.comments}</p> <hr class="solid" />`;

Reply:
using the template literal